cmd/snap/cmd_quota.go

```diff
@@ -165,6 +174,32 @@ func parseCpuQuota(cpuMax string) (count int, percentage int, err error) {
 	return count, percentage, nil
 }
 
+func parseJournalRateQuota(journalRateLimit string) (count int, period time.Duration, err error) {
+	// the rate limit is a string of the form N/P, where N is the number of
+	// messages and P is the period as a time string (e.g 5s)
+	parts := strings.Split(journalRateLimit, "/")
+	if len(parts) != 2 {
+		return 0, 0, fmt.Errorf("missing number of messages and period")
+	}
+
+	count, err = strconv.Atoi(parts[0])
+	if err != nil {
+		return 0, 0, err
+	}
+	if count == 0 {
+		return 0, 0, fmt.Errorf("messages count must be larger than 0")
+	}
+
+	period, err = time.ParseDuration(parts[1])
+	if err != nil {
+		return 0, 0, fmt.Errorf("cannot parse pariod: %v", err)
+	}
+	if period == 0 {
```

Comment:
should we do the same microsecond check as the server instead? another approach is not to do a check here and just let the serve give us back the error

Reply:
I like this approach better. We have checks and validation in place for all quota values on the server.